pkg/openfeature/hooks.go

```diff
@@ -65,3 +65,26 @@ func (h HookContext) ProviderMetadata() Metadata {
 func (h HookContext) EvaluationContext() EvaluationContext {
 	return h.evaluationContext
 }
+
+// check at compile time that UnimplementedHook implements the Hook interface
+var _ Hook = UnimplementedHook{}
+
+// UnimplementedHook implements all hook methods with empty functions
+// Include UnimplementedHook in your hook struct to avoid defining empty functions
+// e.g.
+// type MyHook struct {
+//   UnimplementedHook
+// }
+type UnimplementedHook struct{}
+
+func (u UnimplementedHook) Before(hookContext HookContext, hookHints HookHints) (*EvaluationContext, error) {
+	return nil, nil
```

Comment:
I could be wrong - but will this set the `EvaluationContext` to `nil` before it is used for flag evaluation?

Reply:
Could you add a test for this?

Reply:
Incorrect. The resultant context is only used if not `nil`
Will add a test

Reply:
Added test